shared/js/media/media_frame.js

```diff
@@ -137,6 +139,26 @@ MediaFrame.prototype.displayImage = function displayImage(blob,
   // It would be nice if users of this module could override this
   // background color.
   this.image.style.backgroundColor = '#222';
+  this.image.setAttribute('role', 'img');
+
+  this._localize = function _localize() {
+    var label;
+    var orientation = rotation === 90 || rotation === 270 ?
+      'orientationPortrait' : 'orientationLandscape';
+    if (timestamp) {
+      var locale_entry = navigator.mozL10n.get('imageDescription',
+        { orientation: navigator.mozL10n.get(orientation) });
+      label = MediaUtils.formatDate(timestamp, locale_entry);
+    } else {
+      label = navigator.mozL10n.get('imageDescriptionNoTimestamp',
+        { orientation: navigator.mozL10n.get(orientation) });
+    }
+    this.image.setAttribute('aria-label', label);
+  };
```

Comment:
I will initialize the label to an empty string, but really that isn't necessary either. If there is no locale entry get() returns an empty string, and so does DateTimeFormat.locateFormat().